tornado/web.py

```diff
@@ -2809,6 +2809,23 @@ def validate_absolute_path(self, root: str, absolute_path: str) -> Optional[str]
                 self.redirect(self.request.path + "/", permanent=True)
                 return None
             absolute_path = os.path.join(absolute_path, self.default_filename)
+        else:
+            canonical_path = os.path.normcase(os.path.normpath(absolute_path))
+            new_url = os.path.relpath(canonical_path, self.root).replace(os.path.sep, '/')
+            if os.path.join(self.root, self.path).replace('/', os.path.sep)\
+                    != canonical_path or not self.request.path.endswith(new_url):
+                if not new_url.startswith('/'):
+                    new_url = '/' + new_url
+                if self.request.path.endswith(new_url):
+                    path_prefix = self.request.path[:-len(new_url)]
+                elif urllib.parse.unquote(self.request.path).endswith(new_url):
```

Comment:
This use of `unquote` hard-codes the use of utf8. Does this need to either be configurable or use `unquote_to_bytes`? I can't remember whether StaticFileHandler currently assumes utf8 or works in bytes.

Reply:
No clue where you configure that.